lib/src/rules/require_trailing_commas.dart

```diff
@@ -163,6 +163,15 @@ class _Visitor extends SimpleAstVisitor<void> {
       return true;
     }
 
+    // Exception is allowed if the last parameter is a anonymous function call.
+    // This case arises a lot in asserts.
+    if (lastNode is FunctionExpressionInvocation &&
+        lastNode.function is FunctionExpression &&
+        _isSameLine(lastNode.argumentList.leftParenthesis,
```

Comment:
I'm not sure why this is an interesting condition. If they aren't on the same line then I'd expect the comma would want to go inside the argument list, not outside:
```
  f((a, b, c) {} (
    a,
    b,
    c,
  ));
```

But I did notice that there aren't any tests that cover the case where the parens aren't on the same line.

Reply:
This kind of code looks similar to a function call with an object creation:

```dart
f(MyObject(
  a,
  b,
));
```
And in this case the current version of the lint ([that is really controversal](https://github.com/dart-lang/linter/pull/2566)) is to required a trailing comma to end up to:
```dart
f(
  MyObject(
    a,
    b,
  ),
);
```

So I didn't add anything special here.

> But I did notice that there aren't any tests that cover the case where the parens aren't on the same line.

Added.